core/domain/classifier_services.py

```diff
@@ -153,6 +153,75 @@ def handle_non_retrainable_states(exploration, state_names, exp_versions_diff):
         job_exploration_mappings)
 
 
+FLOAT_INDICATOR_KEY = 'float_values'
+# pylint: disable=too-many-branches
+
+
+def convert_strings_to_float_numbers_in_classifier_data(classifier_data):
+    """Converts all floating point numbers in classifier data to string.
+
+    Args:
+        classifier_data: dict|list. The trained classifier model in which float
+            values are stored as strings.
+
+    Returns:
+        Dict. Original classifier data dict with float values converted back
+            from string to float.
+    """
+    if isinstance(classifier_data, dict):
+        if not FLOAT_INDICATOR_KEY in classifier_data:
```

Comment:
I think you still need to recurse. What if subdicts have floats in them, although the top-level does not?

Reply:
The thing is encoding algorithm adds a `'float_values': []` parameter when no parameter of current dict needs recursion. Which means we can still reach to subdicts. The reason I did this is to ensure that encoding algorithm is indeed travelling through all of the dicts. So we can be sure that current dict has undergone the check by encoding algo even if it didn't make a change. So not having this key means that there is something wrong with encoding algorithm.

Reply:
Oh ok. So then shouldn't we be raising an error here instead?

Reply:
Done.